File: src/R/histoA.R
# Cargar las librerías necesarias
library(ggplot2)
library(readr)

# Leer el archivo CSV
datos <- read_csv("ames-housing.csv", show_col_types = FALSE)

# 1. Histograma
histograma <- ggplot(datos, aes(x = SalePrice)) +
  geom_histogram(binwidth = 10000, fill = "skyblue", color = "black") +
  labs(title = "Histograma de SalePrice",
       x = "Precio de Venta",
       y = "Frecuencia") +
  theme_minimal()

# 2. Boxplot
boxplot <- ggplot(datos, aes(y = SalePrice)) +
  geom_boxplot(fill = "lightgreen") +
  labs(title = "Boxplot de SalePrice",
       y = "Precio de Venta") +
  theme_minimal()

# 3. Violin plot
violin_plot <- ggplot(datos, aes(x = "", y = SalePrice)) +
  geom_violin(fill = "lightpink") +
  labs(title = "Violin Plot de SalePrice",
       x = "",
       y = "Precio de Venta") +
  theme_minimal()

# Mostrar los gráficos
print(histograma)
print(boxplot)
print(violin_plot)

# Opcional: Guardar los gráficos como archivos PNG
ggsave("histograma_saleprice.png", histograma, width = 8, height = 6)
ggsave("boxplot_saleprice.png", boxplot, width = 8, height = 6)
ggsave("violinplot_saleprice.png", violin_plot, width = 8, height = 6)

File: src/histoA.py
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Cargar los datos
df = pd.read_csv('ames-housing.csv')

# Crear una figura con 3 subplots
fig, (ax1, ax2, ax3) = plt.subplots(1, 3, figsize=(20, 6))

# Histograma
sns.histplot(data=df, x='SalePrice', kde=False, ax=ax1)
ax1.set_title('Histograma de SalePrice')
ax1.set_xlabel('Precio de Venta')
ax1.set_ylabel('Frecuencia')

# Boxplot
sns.boxplot(y=df['SalePrice'], ax=ax2)
ax2.set_title('Boxplot de SalePrice')
ax2.set_ylabel('Precio de Venta')

# Violin plot
sns.violinplot(y=df['SalePrice'], ax=ax3)
ax3.set_title('Violin Plot de SalePrice')
ax3.set_ylabel('Precio de Venta')

plt.tight_layout()

# Guardar cada plot como PNG
fig.savefig('histograma_saleprice.png', dpi=300)  # Guardar la figura completa

# Guardar subplots individuales si es necesario
ax1.figure.savefig('histograma_saleprice.png', dpi=300)
ax2.figure.savefig('boxplot_saleprice.png', dpi=300)
ax3.figure.savefig('violin_plot_saleprice.png', dpi=300)

# Ajustar el diseño y mostrar la figura
plt.show()
